app/assets/v2/js/grants/_new.js

```diff
@@ -277,7 +277,22 @@ Vue.mixin({
           _alert(err.message, 'danger');
         }
       });
-    }
+    },
+    handleTwitterUsername(event) {
+      const inputField = event.target;
+      const fullTwitterURL = /https:\/\/twitter.com\/\w{1,15}/
+      const twitterUsername = /(\w{1,15})$/
+
+      let extracted
+
+      if (inputField.value.match(fullTwitterURL)) {
+        extracted = '@' + inputField.value.match(twitterUsername)[0]
```

Comment:
I think you can do all this in one go using a capture group for the username.

```
const  matchResult = inputField.value.match(/https:\/\/twitter.com\/(\w{1,15}/)/)
if (matchResult) { extracted = `@{matchResult[1]}` }
```

I don't think you need two separate regexps to do this.

Reply:
Oh nice. That's really slick